src/main.js

```diff
@@ -16,20 +17,45 @@ const browserHistory = useRouterHistory(createBrowserHistory)({
 // react-router-redux reducer under the key "router" in src/routes/index.js,
 // so we need to provide a custom `selectLocationState` to inform
 // react-router-redux of its location.
-const initialState = window.__INITIAL_STATE__
-const store = configureStore(initialState, browserHistory)
+const store = createStore(window.__INITIAL_STATE__, browserHistory)
 const history = syncHistoryWithStore(browserHistory, store, {
   selectLocationState: (state) => state.router
 })
 
-// Now that we have the Redux store, we can create our routes. We provide
-// the store to the route definitions so that routes have access to it for
-// hooks such as `onEnter`.
-const routes = makeRoutes(store)
-
-// Now that redux and react-router have been configured, we can render the
-// React application to the DOM!
-ReactDOM.render(
-  <Root history={history} routes={routes} store={store} />,
-  document.getElementById('root')
-)
+let render = (key = null) => {
+  const routes = require('./routes/index').default(store)
+  const App = (
+    <Provider store={store}>
+      <div style={{ height: '100%' }}>
+        <Router history={history} children={routes} key={key} />
+      </div>
+    </Provider>
+  )
+  ReactDOM.render(App, MOUNT_ELEMENT)
+}
+
+// Enable HMR and catch runtime errors in RedBox
+// This code is excluded from production bundle
+if (__DEV__ && module.hot) {
+  const renderApp = render
+  const renderError = (error) => {
+    const RedBox = require('redbox-react')
+
+    ReactDOM.render(<RedBox error={error}/>, MOUNT_ELEMENT)
+  }
+  render = () => {
+    try {
+      renderApp(Math.random())
+    } catch (error) {
+      renderError(error)
+    }
+  }
+  module.hot.accept(['./routes/index'], () => render())
+}
+
+// Use Redux DevTools chrome extension
+if (__DEBUG__) {
+  if (!window.devToolsExtension) window.devToolsExtension.open()
```

Comment:
@justingreenberg this logic is actually flipped. It tries to open the extension if it doesn't exist.